lib/pmap.dart

```diff
@@ -1,109 +1,178 @@
 import 'dart:async';
 import 'dart:isolate';
 
-class _Enumerated<T> {
-  final int index;
-  final T value;
-  _Enumerated({this.index, this.value});
+/// The *main* process for the spawned isolates.
+void _process(_Processor processor) async {
+  final receivePort = ReceivePort();
+  processor.sendPort.send(receivePort.sendPort);
+  await for (dynamic input in receivePort) {
+    processor.process(input);
+  }
 }
 
+/// This [_Processor] is given to the spawned isolates as the work basis.
```

Comment:
How about:
/// An object that represents all the data an isolate needs to perform its work.